cumulusci/cli/cli.py

```diff
@@ -305,115 +317,153 @@ def service(config):
-              help="The tag prefix for production release tags",
-              prompt=True,
-              default=CLI_CONFIG.global_config.project__git__prefix_release,
-              )
-@click.option('--test-name-match',
-              help="The SOQL format like query for selecting Apex tests.  % is wildcard",
-              prompt=True,
-              default=CLI_CONFIG.global_config.project__test__name_match,
-              )
+@click.option('--extend', help="If set to the url of another Github repository configured for CumulusCI, creates this project as an extension which depends on the other Github project and all its dependencies")
 @pass_config
-def project_init(config, name, package_name, package_namespace, package_api_version, git_prefix_feature, git_default_branch, git_prefix_beta, git_prefix_release, test_name_match):
+def project_init(config, extend):
     if not os.path.isdir('.git'):
-        click.echo("You are not in the root of a Git repository")
+        raise click.ClickException("You are not in the root of a Git repository")
 
     if os.path.isfile('cumulusci.yml'):
-        click.echo("This project already has a cumulusci.yml file")
+        raise click.ClickException("This project already has a cumulusci.yml file")
 
     yml_config = []
-    # project:
+
+    click.echo()
+    click.echo(click.style('# Project Info', bold=True, fg='blue'))
+    click.echo('The following prompts will collect general information about the project')
+    package_config = []
+
+    name = os.path.split(os.getcwd())[-1:][0]
+    click.echo()
+    click.echo('Enter the project name.  The name is usually the same as your repository name')
+    name = click.prompt(click.style('Project Name', bold=True), default=name)
     yml_config.append('project:')
     yml_config.append('    name: {}'.format(name))
 
-    #     package:
-    package_config = []
+    click.echo()
+    click.echo("CumulusCI uses an unmanaged package as a container for your project's metadata.  Enter the name of the package you want to use.")
+    package_name = click.prompt(click.style('Package Name', bold=True), default=name)
     if package_name and package_name != config.global_config.project__package__name:
         package_config.append('        name: {}'.format(package_name))
-    if package_namespace and package_namespace != config.global_config.project__package__namespace:
-        package_config.append(
-            '        namespace: {}'.format(package_namespace))
+
+    click.echo()
+    package_namespace = None
+    if click.confirm(click.style("Is this a managed package project?", bold=True), default=False):
+        click.echo('Enter the namespace assigned to the managed package for this project')
+        package_name = click.prompt(click.style('Package Namespace', bold=True), default=name)
+        if package_namespace and package_namespace != config.global_config.project__package__namespace:
+            package_config.append(
```

Comment:
I have to ask if theres a way to build this config in memory as a dict and just dump to yaml?

Reply:
i know the yaml strings are't new but...

Reply:
The yaml dump doesn't respect ordereddict and even the ordereddict dumper in HiYaPyCo which we have available doesn't handle recursive dictionaries very well.  I do want to revisit this at some point, probably with building a model of some sort for the cci config and then being able to render that as json, yaml, etc.   We could use that model and its validation in a bunch of places.

Reply:
marshmallow!